typescriptSrc/world.ts

```diff
@@ -72,166 +72,177 @@ module world {
+              const val = new StringV(result+"");
+              vms.finishStep(val);
+          }
+        }
+    }
+
+    function comparatorStepperFactory(callback: (vals: Array<number>) => boolean): (vms: VMS, args: Array<Value>) => void {
+      return function(vms : VMS, args : Array<Value>) : void {
+        const vals : Array<number>= [] ;
+        let ok = true ;
+        for( let i=0 ; i < args.length ; ++i ) {
+            if( canConvertToNumber( args[i] ) ) {
+                vals.push( convertToNumber( args[i] ) ) ; }
+            else {
+                vms.reportError( "The "+nth(i+1)+" argument is not a number.");
+                ok = false ; } }
+        
+        if( ok ) {
+            const result = callback(vals);
+            const val = new StringV( result+"" );
+            vms.finishStep( val ) ;
+        }
+      }
+    }
+
+    function logicalStepperFactory(callback: (leftOperand: boolean, rightOperand: boolean) => boolean): (vms: VMS, args: Array<Value>) => void {
+      return function andstep( vms : VMS, args : Array<Value> ) : void {
+        const vals : Array<boolean>= [] ;
+        let ok = true ;
+        for( let i=0 ; i < args.length ; ++i ) {
+            if(isBool(args[i])) {
+              vals.push(convertToBool(args[i]));                    
+            } else {
+              vms.reportError("The "+nth(i+1)+" argument is not a bool.");
+              ok = false;
+            }
+        }    
+        if(ok) {
+            const result = vals.reduce(callback);
+            const val = new StringV(result+"") ;
+            vms.finishStep(val);
+        }
+      }
+    }
+
     export class World extends ObjectV {
-        protected stepperFactory: {[value: string]: (vms : VMS) => void;};
 
         constructor() {
             super();
             //console.log("World's fields array is length: " + this.fields.length);
 
-            this.stepperFactory = {};
-
-            function addstep(vms: VMS) : void {
-              const node = vms.getPendingNode();
-              const vals : Array<number>= [];
-              let ok = true;
-              if (node.count() > 0) {
-                  for (let i = 0; i < node.count(); i++) {
-                      let childVal = vms.getChildVal(i);
-                      if (canConvertToNumber(childVal)) {
-                          vals.push(convertToNumber(childVal))
-                      } else {
-                          vms.reportError( "The "+nth(i+1)+" argument is not a number.");
-                          ok = false; 
-                      } 
-                  }
-              }  
-              if(ok) {
-                  const sum = vals.reduce( (s, x) => s+x, 0 ) ;
-                  const val = new StringV( sum+"" ) ;
-                  vms.finishStep( val ) ;
-              }
-            }
-
+            let addCallback = function(leftOperand: number, rightOperand: number): number { return leftOperand + rightOperand; } 
```

Comment:
It's not a big deal, but the coding standard for the project says to prefer fat-arrow lambdas.